### DISCUSSION TITLE
Will we ever get local run in browser?

### DISCUSSION THREAD

**Discussion by TheThornZ** [ruffle-rs/ruffle#3008] [General]
Will we ever get local run in browser?
Basically I want to be able to run an SWF file locally within a browser with all files hosted on the machine itself. 

Reason: would like to get this 8 year old tutorial back up and running.
https://youtu.be/t0VxpwDZW54

![screenshot](https://user-images.githubusercontent.com/77697874/106398925-9b00da00-63ca-11eb-9300-36dcff511c6b.PNG)

**Comment by ousia** [comment]
@TheThornZ,

I have just checked it and it works fine (at least on Firefox).

Remote site:

![remote](https://user-images.githubusercontent.com/203767/106399570-bc3ce700-6419-11eb-8506-7266a5384a12.png)

Local site (too lazy to copy style sheets and such):

![local](https://user-images.githubusercontent.com/203767/106399580-d24aa780-6419-11eb-973d-b1a170d53e32.png)

Update extension to latest version and open an issue with a minimal sample.

I don’t know whether Chrome might be too picky about running local files.

You might install a developer version of Firefox (even a [portable one](https://portableapps.com/apps/internet/firefox-developer-portable)).

Just in case it might help.

**Reply by TheThornZ** [reply]
Thanks for sharing, unfortunately the target is Chrome, more specifically running local on a Chromebook. The tutorial I wrote was to run the freeware/shareware versions of quake and doom on an offline Chromebook. Since both were flash based they have since been rendered dead. I did however manage to get an HTML5 version of Wolfenstein 3D so that's still working. Was just hopeful I could have used ruffle rather than starting from square one and redoing the projects in HTML5.

**Comment by n0samu** [comment]
The short answer is no, Ruffle extension/selfhosted builds cannot work with local files. [See the FAQ](https://github.com/ruffle-rs/ruffle/wiki/Frequently-Asked-Questions-For-Users#how-can-i-use-ruffle-with-local-html-files).
You can use the Ruffle Web Demo to play individual Flash files on a Chromebook, or you can host the files on a website or local web server.

**Comment by lvyitian** [comment]
The wasm requires a server to load it out of the security policy.
